src/browser/plugins.ts

```diff
@@ -0,0 +1,73 @@
+/*
+Copyright 2017 OpenFin Inc.
+
+Licensed under OpenFin Commercial License you may not use this file except in compliance with your Commercial License.
+Please contact OpenFin Inc. at sales@openfin.co to obtain a Commercial License.
+*/
+
+import * as path from 'path';
+import { getStartManifest, StartManifest } from './core_state';
+import { Plugin } from '../shapes';
+import { readFile } from 'fs';
+import { rvmMessageBus } from './rvm/rvm_message_bus';
+
+interface PluginWithContent extends Plugin {
+    _content: string;
+}
+
+/**
+ * A map of already-retrieved plugin module paths.
+ *
+ * Examples:
+ * 'pluginName: 1.0.0' -> 'path\to\plugin\injectable\script.js'
+ * 'pluginName2: 0.0.1' -> '' // plugin/version doesn't exist
+ */
+const pluginPaths: Map<string, string> = new Map();
+
+/**
+ * Gets all plugins defined in app's manifest
+ */
+export async function getModules(): Promise<PluginWithContent[]> {
```

Comment:
For all of these `async` calls, if the only place they are ever called is form the api_handler then perhaps not handling the fails is OK as that is explicitly try caught, but we need to be sure that these cannot blow up browser side outside a try/catch. This may already be the case.

Reply:
Should be fine / okay

Reply:
:+1: